File: tests/steps/1/step1.java
import org.omg.CORBA.*;
import org.omg.CORBA.ORBPackage.*;
import ua.kiev.gradsoft.UAKGQuery.*;

public class step1
{
    public static void main(String args[])
    {
        try
        {
            ORB orb = ORB.init(args, props);
            org.omg.CORBA.Object obj = null;
            DBConnectionManager dbcm = null;
            try
            {
                obj = orb.resolve_initial_references("UAKGQueryService");
            }
            catch(InvalidName ex)
            {
                System.out.println("UAKGQueryService service is not available");
                return;
            }
            if(obj == null)
            {
                System.out.println("UAKGQueryService object reference is invalid");
                return;
            }
            dbcm = DBConnectionManagerHelper.narrow(obj);
            if(dbcm == null)
            {
                System.out.println("Can't narrow DBConnection to correct type");
                return;
            }                                              
            UAKGQueryManager queryManager;
            try 
            {
                queryManager = dbcm.createQueryManager("scott", "tiger", "", "Oracle8", "");
            }
            catch(QueryManagerNotFound ex)
            {
                System.out.println("Can't find query manager.");
                return;
            }
        }
        catch(SystemException ex)
        {
            System.out.println("SystemException" + ex.getMessage());
            return;
        }
        System.out.println("Success!!!");
    }
}

File: tests/steps/1/step1.cpp
#include <QueryStepsPostConfig.h>

#include <importCORBA.h>
#include <importUAKGQuery.h>
#include <UAKGQuery2Init.h>

#include <iostream>

using namespace std;

int main(int argc, char** argv)
{
  CORBA::ORB_var orb;  
  try {


#ifdef CORBA_ORB_INIT_HAVE_3_ARGS
    orb = CORBA::ORB_init(argc, argv, CORBA_ORB_INIT_THIRD_ARG);
#else
    orb = CORBA::ORB_init(argc, argv);
#endif

    initUAKGQuery2(orb); /* register valuetyre */

    CORBA::Object_var obj;
    try {
       obj = orb->resolve_initial_references("UAKGQueryService");
    }catch(const CORBA::ORB::InvalidName&){
       std::cerr << argv[0] << ": can't resolve UAKGQueryService" << std::endl;
       return 1;
    }
        
    if(CORBA::is_nil(obj)) {
      std::cerr << argv[0] << 
                   ": UAKGQueryService is a nil object reference" << std::endl;
      return 1;
    }

    UAKGQuery2::DBConnectionManager_var dbManager = 
                                UAKGQuery2::DBConnectionManager::_narrow(obj);

    if (CORBA::is_nil(dbManager)) {
      std::cerr << argv[0] << 
                  ": can't narrow dbManager to correct type" << std::endl;
      return 1;
    } 
   
    UAKGQuery2::QueryManager_var queryManager;

    try {
      queryManager = 
           dbManager->create_query_manager("skott","tiger",
                                           "","Oracle8","");
    }catch(const UAKGQuery2::QueryManagerNotFound& ex){
      std::cerr << argv[0] <<": can't find query manager." << std::endl;
      return 1;
    }
    
   // here you can perform 


    //time to disconnect.
    queryManager->destroy();
    std::cout << "disconnected." << std::endl;
  }catch(const CORBA::SystemException& ex){
    cerr << ex << endl;

    orb->destroy(); 
    return 1;
  }
  orb->destroy(); 
  return 0;
}

